### DISCUSSION TITLE
aborting LTSP, while booting client on debian server

### DISCUSSION THREAD

**Discussion by jorigel** [ltsp/ltsp#388] [General] [answered]
Hi,
newly my system stops while booting showing the error:

ltsp command failed: chroot /root rsync -a --delete /tmp/ /usr/share/ltsp
picture complete error list:[https://servmin.duckdns.org/paste/f.php?h=3OpdkgsU&p=1](url)

What information is needed to solve  this problem and what to do?
thanX for Ur attention
jori

**Comment by alkisg** [comment]
Hi, you need a fixed rsync. Which client/rsync version is this? I've only seen this in Ubuntu 20.10.
If it happens in Debian Bullseye, please open a bug report in Debian and point them to the following bug reports that I filed:

https://bugs.launchpad.net/ubuntu/+source/rsync/+bug/1902109
https://github.com/WayneD/rsync/issues/109

**Reply by jorigel** [reply]
Hi Alkis,
it is an update to ubuntu 20.10, indeed. It's not worth to make a big deal out of it, because it's only 2 month away from the next lts-version. I'm going back to 20.04 and shall try the mate-desktop that You are using and mentioned in the installation description of Your ltsp-pages.

Doing that, I noticed that the ltsp ipxe command doesn't result in a working menu with different entries as I expected. Staying with one entry even when a couple of images are present is normal and like it should be?
The ltsp ipxe man page is not, what really helped me to understand how this thing is working. Is there a place where a non-insider can get into it or is there a short advice You could give me to get a menu with the expected entries?

thanks a lot for Your help and stay healthy
jori

ps.: oh, should I have opened a new post for this? :-o, sorry . . .

**Comment by alkisg** [comment]
> it's only 2 month away from the next lts-version.

Note that 21.04 is NOT an LTS version. The next LTS version will be Ubuntu 22.04.

> Staying with one entry even when a couple of images are present is normal?

No, it's not normal. Paste the output of the following commands:

```shell
ltsp ipxe
ls -l /srv/ltsp/images
cat /srv/tftp/ltsp/ltsp.ipxe
```

**Reply by jorigel** [reply]
Hi,
sorry, quick shot, lts is next year, of course -> my fault :)

the result of the commands:

```shell
ltsp ipxe ->
Installed /usr/share/ltsp/server/ipxe/ltsp.ipxe in /srv/tftp/ltsp/ltsp.ipxe
Skipped existing /srv/tftp/ltsp/memtest.0
Skipped existing /srv/tftp/ltsp/memtest.efi
Skipped existing /srv/tftp/ltsp/snponly.efi
Skipped existing /srv/tftp/ltsp/undionly.kpxe

ls -l /srv/ltsp/images ->
insgesamt 7726580
-rw-r--r-- 1 root root 1993846784 Jan 27 18:35 ubuntu-mate-20.04.01.img
-rw-r--r-- 1 root root 1850966016 Jan 26 02:13 xubuntu-groovy.img
-rw-r--r-- 1 root root 4067192832 Jan 26 02:38 xubuntu.img

cat /srv/tftp/ltsp/ltsp.ipxe ->
#!ipxe
# This file is part of LTSP, https://ltsp.org
# Copyright 2019 the LTSP team, see AUTHORS
# SPDX-License-Identifier: GPL-3.0-or-later

# Configure iPXE for LTSP
# Documentation=man:ltsp-ipxe(8)

# Set the default image (img) based on arch, or to root-path if it's not empty
cpuid --ext 29 && set img x86_64 || set img x86_32
isset ${root-path} && set img ${root-path} ||
isset ${proxydhcp/dhcp-server} && set srv ${proxydhcp/dhcp-server} || set srv ${next-server}

# Define cmdline parameters common to all LTSP clients
set cmdline_ltsp BOOTIF=01-${mac:hexhyp}

goto ${mac} || goto start
# The following client-specific settings can be defined in ltsp.conf:
# DEFAULT_IMAGE (img): default menu item for that client
# KERNEL_PARAMETERS (cmdline_client): additional kernel parameters
# MENU_TIMEOUT (menu-timeout): menu milliseconds, 0=forever, -1=hide menu
# For those clients, :mac:address sections are generated below

:61:6c:6b:69:73:67

:start
# To completely hide the menu, set menu-timeout to -1
isset ${menu-timeout} || set menu-timeout 5000
iseq "${menu-timeout}" "-1" && goto ${img} ||
menu iPXE boot menu - ${hostname}:${srv}:${root-path} || goto ${img}
item --gap                        Boot an image from the network in LTSP mode:
item --key 3 xubuntu-groovy       xubuntu-groovy.img

item
item --gap                        Other options:
item --key m memtest              Memory test
item --key c config               Enter iPXE configuration
item --key s shell                Drop to iPXE shell
item --key d disk                 Boot from the first local disk
item
item --key x exit                 Exit iPXE and continue BIOS boot
choose --timeout ${menu-timeout} --default ${img} img || goto cancel
goto ${img}

:xubuntu-groovy
:xubuntu
:ubuntu-mate-20.04.01
:images
# The "images" method can boot anything in /srv/ltsp/images
set cmdline_method root=/dev/nfs nfsroot=${srv}:/srv/ltsp ltsp.image=images/${img}.img loop.max_part=9
goto ltsp

:roots
# The "roots" method can boot all /srv/ltsp/roots
set cmdline_method root=/dev/nfs nfsroot=${srv}:/srv/ltsp/${img}
goto ltsp

:ltsp
# :images and :roots jump here after setting cmdline_method
set cmdline ${cmdline_method} ${cmdline_ltsp} ${cmdline_client}
# In EFI mode, iPXE requires initrds to be specified in the cmdline
kernel /ltsp/${img}/vmlinuz initrd=ltsp.img initrd=initrd.img ${cmdline}
initrd /ltsp/ltsp.img
initrd /ltsp/${img}/initrd.img
boot || goto failed

:memtest
iseq ${platform} pcbios && kernel memtest.0 || kernel memtest.efi
# Boot "fails" on normal memtest exit with Esc, so show the menu again
boot ||
goto start

:config
config
goto start

:shell
echo Type 'exit' to get the back to the menu
shell
goto start

:disk
# Boot the first local HDD
sanboot --no-describe --drive 0x80 || goto failed

:exit
# Exit with error, to fall back to the next boot option
# http://forum.ipxe.org/showthread.php?tid=6775
exit 1

:cancel
echo You cancelled the menu, dropping to shell
goto shell

:failed
echo Booting failed, dropping to shell
goto shell
```

and the result of it, is a menu with only one entry like under the start: section

best regards
jori

**Comment by alkisg** [comment]
How were these images created, with `ltsp image name` or by just copying an .img file?
If you just copy the .img there, you'll also need to run `ltsp kernel name` and then `ltsp ipxe`, in order for it to appear to the ipxe menu.

**Reply by jorigel** [reply]
That was the way I thought it should work.
So in the beginning I created links from the virtualbox directory to the ltsp one.
To show You what I then did I'm doing it again here:

[commands.txt](https://github.com/ltsp/ltsp/files/5886926/commands.txt)

and as You see, the result seems to be the same. It has never been different to me since the beginning. It looks like as if I'm doing something wrong. Do You see the reason for this  unexpected result?

**Comment by alkisg** [comment]
You're doing things correctly, yet the result is semi-wrong.
It's placing the images as goto labels, but not as menu items.

Did you happen to manually modify the /usr/share/ltsp/server/ipxe/ltsp.ipxe template at some point?
Can you paste all of it here?
Also, which ltsp version is that? `ltsp -V`

**Reply by jorigel** [reply]
Hi,
here's the ltsp Version running on debian server as You see:
```
ltsp -V ->
ltsp 21.01-1~ubuntu20.04.1

lsb_release -a ->
No LSB modules are available.
Distributor ID:	Debian
Description:	Debian GNU/Linux 10 (buster)
Release:	10
Codename:	buster

```

I don't remember and I don't have a protocol of changing /usr/share/ltsp/server/ipxe/ltsp.ipxe. The only thing I did two days ago to solve the problem of interrupting image load was to reinstall the complete ltsp like this:

```
apt reinstall ltsp ltsp-binaries dnsmasq nfs-kernel-server openssh-server squashfs-tools ethtool net-tools epoptes

ltsp dnsmasq


```
I didn't purge the old installation. So may be old versions of configuration files remained?
The contents of the requested file is:

```
cat /usr/share/ltsp/server/ipxe/ltsp.ipxe ->
#!ipxe
# This file is part of LTSP, https://ltsp.org
# Copyright 2019 the LTSP team, see AUTHORS
# SPDX-License-Identifier: GPL-3.0-or-later

# Configure iPXE for LTSP
# Documentation=man:ltsp-ipxe(8)

# Set the default image (img) based on arch, or to root-path if it's not empty
cpuid --ext 29 && set img x86_64 || set img x86_32
isset ${root-path} && set img ${root-path} ||
isset ${proxydhcp/dhcp-server} && set srv ${proxydhcp/dhcp-server} || set srv ${next-server}

# Define cmdline parameters common to all LTSP clients
set cmdline_ltsp BOOTIF=01-${mac:hexhyp}

goto ${mac} || goto start
# The following client-specific settings can be defined in ltsp.conf:
# DEFAULT_IMAGE (img): default menu item for that client
# KERNEL_PARAMETERS (cmdline_client): additional kernel parameters
# MENU_TIMEOUT (menu-timeout): menu milliseconds, 0=forever, -1=hide menu
# For those clients, :mac:address sections are generated below

:61:6c:6b:69:73:67

:start
# To completely hide the menu, set menu-timeout to -1
isset ${menu-timeout} || set menu-timeout 5000
iseq "${menu-timeout}" "-1" && goto ${img} ||
menu iPXE boot menu - ${hostname}:${srv}:${root-path} || goto ${img}
item --gap                        Boot an image from the network in LTSP mode:
# item --key 1 images             x86_64
item
item --gap                        Other options:
item --key m memtest              Memory test
item --key c config               Enter iPXE configuration
item --key s shell                Drop to iPXE shell
item --key d disk                 Boot from the first local disk
item
item --key x exit                 Exit iPXE and continue BIOS boot
choose --timeout ${menu-timeout} --default ${img} img || goto cancel
goto ${img}

:images
# The "images" method can boot anything in /srv/ltsp/images
set cmdline_method root=/dev/nfs nfsroot=${srv}:/srv/ltsp ltsp.image=images/${img}.img loop.max_part=9
goto ltsp

:roots
# The "roots" method can boot all /srv/ltsp/roots
set cmdline_method root=/dev/nfs nfsroot=${srv}:/srv/ltsp/${img}
goto ltsp

:ltsp
# :images and :roots jump here after setting cmdline_method
set cmdline ${cmdline_method} ${cmdline_ltsp} ${cmdline_client}
# In EFI mode, iPXE requires initrds to be specified in the cmdline
kernel /ltsp/${img}/vmlinuz initrd=ltsp.img initrd=initrd.img ${cmdline}
initrd /ltsp/ltsp.img
initrd /ltsp/${img}/initrd.img
boot || goto failed

:memtest
iseq ${platform} pcbios && kernel memtest.0 || kernel memtest.efi
# Boot "fails" on normal memtest exit with Esc, so show the menu again
boot ||
goto start

:config
config
goto start

:shell
echo Type 'exit' to get the back to the menu
shell
goto start

:disk
# Boot the first local HDD
sanboot --no-describe --drive 0x80 || goto failed

:exit
# Exit with error, to fall back to the next boot option
# http://forum.ipxe.org/showthread.php?tid=6775
exit 1

:cancel
echo You cancelled the menu, dropping to shell
goto shell

:failed
echo Booting failed, dropping to shell
goto shell

```

Hope that helps. B.t.w. thank You for Your attention Alkis.
jori

**Comment by alkisg** [comment]
At that point I can only imagine weird things like the use of `chattr` to prevent overwrites etc.
The easiest way to troubleshoot the problem would be IRC+VNC. If you'd like that, go to ltsp.org and click on the IRC link.

**Comment by alkisg** [comment]
The problem was that /bin/sh was a symlink to /bin/bash, and that `ltsp ipxe` has an issue with bash.
Filed #389 about it, thanks.

**Comment by alkisg** [comment, marked as answer]
Ubuntu 21.04 still has the issue with "rsync: failed setting permissions...".
A workaround is to put the following in ltsp.conf, under [clients]:

```shell
PRE_INITRD_BOTTOM_RSYNC='chroot() { re mount --bind /proc "$1/proc"; re /bin/chroot "$@"; re umount "$1/proc"; }'
```

**Reply by Purpi** [reply]
Hello,

I ran into the rsync issue. I was able to boot the client using the workaround mentioned above. Also the solution with an old version of rsync worked.
But the users can't log in. Neither with the graphical login, nor in the console. The error message is "login incorrect". But the password is correct for sure.

Can anybody help me? Thanks in advance!


Greetings
Marcus

**Reply by Saehing** [reply]
> Ubuntu 21.04 still has the issue with "rsync: failed setting permissions...". A workaround is to put the following in ltsp.conf, under [clients]:
> 
> ```shell
> PRE_INITRD_BOTTOM_RSYNC='chroot() { re mount --bind /proc "$1/proc"; re /bin/chroot "$@"; re umount "$1/proc"; }'
> ```

It didn't work, are there any other options?

**Comment by alkisg** [comment]
Hi, see issue #450.

**Comment by alkisg** [comment]
@Saehing, can you upload a photo of the issue?
Maybe you didn't run `ltsp initrd`?
For me, the workaround works for both 21.04 and 21.10.

(btw please don't use inline threads I can't follow them they're confusing me :))
